controllers/tasks.js

```diff
@@ -43,6 +43,27 @@ const fetchTasks = async (req, res) => {
     return res.boom.badImplementation('An internal server error occurred')
   }
 }
+
+const getUserTasks = async (req, res) => {
+  try {
+    const { username } = req.params
+
+    if (username) {
+      const allTasks = await tasks.fetchUserTasks(username, [], true)
```

Comment:
Once we take `username` from query, we should find `userId` corresponding to the given `username` using existing functions from `utils/users.js`.

We are storing `userId` in DB and not `username`

Reply:
We are fetching `userId`  in `fetchUserTasks` function. It was existing function.

Reply:
No. You're passing `username` from here, how will the `fetchUserTasks` function know the corresponding userId accordingly by itself? 😅 
Just to make it clear, my `username` would be `swaraj` and my userId would be some random string generated by firestore and it would be unique and it won't change any time soon. My username may change, but the userId won't and hence we're using `userId` in the DB to store things.